### ISSUE TITLE
Possibly throttle EtherscanClient calls

### ISSUE THREAD

**Issue by eierina** [naddison36/tx2uml#53]
In the EtherscanClient, the getContract and getToken calls can execute multiple times in a short period of time causing the following error on free Etherscan accounts:

```
  [cause]: Error: Failed Etherscan API with result "Max rate limit reached, rate limit of 5/1sec applied"
```

It would be sufficient to throttle the 2 calls at something bit larger than 200ms to allow non paying users to use the (amazing) tool.

**Comment by naddison36** [comment]
Thanks for raising. I throttle getting the Etherscan contracts for the contract call diagrams but I missed this for value transfer diagrams. 
https://github.com/naddison36/tx2uml/blob/90cdd824316e9ecefe773f4956cc81a555112895/src/ts/transaction.ts#L210

Was the problem with just the `value` transfer diagrams? Or also contract `call` diagrams?

**Comment by eierina** [comment]
Not sure which one of the two was, I have the full error message if it helps

Error: Failed to get contract details for contract 0x...from Etherscan using url https://api-optimistic.etherscan.io/api
    at EtherscanClient.getContract (/Users/.../Projects/PERSONAL/tx2uml/lib/clients/EtherscanClient.js:73:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  [cause]: Error: Failed Etherscan API with result "Max rate limit reached, rate limit of 5/1sec applied"
      at EtherscanClient.getContract (/Users/.../Projects/PERSONAL/tx2uml/lib/clients/EtherscanClient.js:41:23)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      
If that does not help I can check for you later.

**Comment by naddison36** [comment]
What was the command used to run so I can replicate? Thanks

**Comment by eierina** [comment]
I believe the issue is in transacton.ts' getContractsFromAddresses in the followin line

const getContractPromises = addresses.map(address => {
            return limit(() => this.etherscanClient.getContract(address))
        })

It builds a map of getContract calls and then executes all at once.

**Comment by naddison36** [comment]
This should be fixed in Release v1.1.17 https://github.com/naddison36/tx2uml/pull/62

**Comment by naddison36** [comment]
v1.1.17 has been released
